qt/scientific_interfaces/Direct/ALFAnalysisModel.cpp

```diff
@@ -152,6 +160,34 @@ IFunction_sptr ALFAnalysisModel::calculateEstimate(MatrixWorkspace_sptr &workspa
   return createCompositeFunction(createFlatBackground(), createGaussian());
 }
 
+void ALFAnalysisModel::exportWorkspaceCopyToADS() const {
+  // The ADS should not be used anywhere else apart from here. Note that a copy is exported.
+  if (auto const workspace = plottedWorkspace()) {
+    AnalysisDataService::Instance().addOrReplace(WS_EXPORT_NAME, workspace->clone());
+  }
+}
+
+void ALFAnalysisModel::openExternalPlot() const {
+  // Externally plot the extracted workspace or fitted workspace depending on which one is available.
+  if (auto const workspace = plottedWorkspace()) {
+    MantidQt::Widgets::MplCpp::plot({workspace}, boost::none, plottedWorkspaceIndices());
```

Comment:
Is the model the right place to be doing the plotting? It feels like it breaks MVP a bit for the model to know about this other Qt widget (which is really a view concern). It also makes it hard to test. 

Would it be worth creating a little helper class like we do [here](https://github.com/mantidproject/mantid/blob/main/qt/scientific_interfaces/ISISReflectometry/GUI/Common/Plotter.h) in the reflectometry GUI? This means we can mock it and it also makes it easy to customise the plot.

Reply:
Good point, I've moved the external plotting code to the view. Has allowed me to add more tests so clearly this is the right thing to do 👍